src/backend/distributed/master/master_truncate.c

```diff
@@ -139,3 +144,75 @@ TruncateTaskList(Oid relationId)
 
 	return taskList;
 }
+
+
+/*
+ * truncate_local_table truncates the local records of a distributed table.
+ *
+ * The main advantage of this function is to truncate all local records after creating a
+ * distributed table, and prevent constraints from failing due to outdated local records.
+ */
+Datum
+truncate_local_table(PG_FUNCTION_ARGS)
+{
+	EnsureCoordinator();
+	Oid relationId = PG_GETARG_OID(0);
+
+	EnsureLocalTableCanBeTruncated(relationId);
+	EnsureTableOwner(relationId);
+
+	TruncateStmt *truncateStmt = makeNode(TruncateStmt);
+
+	char *relationName = get_rel_name(relationId);
+	List *names = stringToQualifiedNameList(relationName);
+	truncateStmt->relations = list_make1(makeRangeVarFromNameList(names));
```

Comment:
Does this work correctly for schemas? requires some test

Reply:
`get_rel_name` definition has [some comments](https://github.com/postgres/postgres/blob/3e62dd3a937b7da84e8ae126c7bf8319c1cd5c08/src/backend/utils/cache/lsyscache.c#L1726-L1727) that imply it may not be safe
> NOTE: since relation name is not unique, be wary of code that uses this for anything except preparing error messages.

I will fix this one

Reply:
replaced `get_rel_name` with `generate_qualified_relation_name` and `stringToQualifiedNameList` calls